### PR TITLE
Update to PHPCS 3.4 and use new Sniffs

### PR DESCRIPTION
**Pull request by carusogabriel**
https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.4.0

### CODE REVIEW

**Comment by Majkl578** [comment]
Can we do new sniffs separately please? :Pray:

**Comment by carusogabriel** [comment]
@Majkl578 One of them is because of depreciation, the other one I doubt someone disagrees with that. But, if it is needed, okay for me.

**Comment by Majkl578** [comment]
- It'ws not obvious see what deprecation is being handled there.
- Updating Slevomat CS dependency is not within scope of "Update PHPCS to 3.4".

File: lib/Doctrine/ruleset.xml [hunk 1]
```diff
@@ -457,6 +461,8 @@
             <property name="spacingAfterLast" value="0"/>
         </properties>
     </rule>
+    <!-- Require no space between increment/decrement operator and its variable -->
```
Comment (greg0ire) [inline lib/Doctrine/ruleset.xml]: "Require no space" can be intepreted as "Do not require a space", let's maybe reword:

```suggestion
    <!-- Require there be no space between increment/decrement operator and its operand -->
```
Also, "variable" => "operand"

**Comment by carusogabriel** [comment]
@greg0ire Thanks!

@Majkl578 I've split the changes in more commits. Let me know what else should I do so we can get this merged 😊
